hibernate-core/src/main/java/org/hibernate/dialect/H2Dialect.java

```diff
@@ -237,60 +236,44 @@ public SqmMultiTableMutationStrategy getFallbackSqmMutationStrategy(
 	}
 
 	@Override
-	public ViolatedConstraintNameExtracter getViolatedConstraintNameExtracter() {
-		return EXTRACTER;
-	}
-
-	private static final ViolatedConstraintNameExtracter EXTRACTER = new TemplatedViolatedConstraintNameExtracter() {
-		/**
-		 * Extract the name of the violated constraint from the given SQLException.
-		 *
-		 * @param sqle The exception that was the result of the constraint violation.
-		 * @return The extracted constraint name.
-		 */
-		@Override
-		protected String doExtractConstraintName(SQLException sqle) throws NumberFormatException {
-			String constraintName = null;
-			// 23000: Check constraint violation: {0}
-			// 23001: Unique index or primary key violation: {0}
-			if ( sqle.getSQLState().startsWith( "23" ) ) {
-				final String message = sqle.getMessage();
-				final int idx = message.indexOf( "violation: " );
-				if ( idx > 0 ) {
-					constraintName = message.substring( idx + "violation: ".length() );
+	public ViolatedConstraintNameExtractor getViolatedConstraintNameExtracter() {
+		return EXTRACTOR;
+	}
+
+	private static final ViolatedConstraintNameExtractor EXTRACTOR =
+			new TemplatedViolatedConstraintNameExtractor( sqle -> {
+				// 23000: Check constraint violation: {0}
+				// 23001: Unique index or primary key violation: {0}
+				if ( sqle.getSQLState().startsWith( "23" ) ) {
+					final String message = sqle.getMessage();
+					final int idx = message.indexOf( "violation: " );
+					if ( idx > 0 ) {
```

Comment:
Just wanna confirm the 'violation: ' never shows up as prefix, right?

Reply:
No clue. Again, not looking to make functional changes with this PR.